[FE] Positions List Search Items

mariano-aguero opened the pull request:
Closes #496 
Closes #413 
Closes #411 

Regarding 413:
- Token address is the same as Collateral Address, I choose only one

Regarding 411:
- Wrapped Collateral: All / Yes / No , we can only check if the Wrapped collateral address is initialized in the subgraph (not null)

gnosis-info commented:
Travis automatic deployment:
 https://pr497--conditionaltokensexplorer.review.gnosisdev.com

gabitoesmiapodo reviewed (changes requested):
#496 is great

#413 I think that by "Token address" I meant "Wrapped collateral address" or "WMT Address" (or I may have meant to use "Collateral Address" for that... we changed some wording here and there, so I could be wrong, it was called WMT too). Anyways: I don't know if it's possible to implement. We can do it later, maybe?

 #411 I think the "Custom Token" option is missing from the "Collateral" filter.

Everything else seems to work great.

gnosis-info commented:
Travis automatic deployment:
 https://pr497--conditionaltokensexplorer.review.gnosisdev.com

elena-zh commented:
@mariano-aguero , 
I also have verified the changes. Great work! However, here are some notes: 
1. #411 - Hiding the filters' column does not clear Wrapped collateral filtering
2. creation date filter shows positions that were created on 'the selected day - 1'
![creation date](https://user-images.githubusercontent.com/70885163/96867303-30f4a280-1475-11eb-84f2-eedbdaf6ad9c.jpg)

3. validation message is displayed when clear out any filter field (both or any of them - a user cannot start a search when the validation message is displayed)
![clear out](https://user-images.githubusercontent.com/70885163/96867312-3520c000-1475-11eb-9f56-fbe16a8d39a0.jpg)

4. It seems to me that search by a custom collateral symbol does not work: see the video https://drive.google.com/file/d/16RCYVtBS93JqkKvvfMlIKWXFi9pCTHDv/view
5. the issue is not exactly related to the current ticket, but it is worth fixing: #453 (incorrect number of positions are displayed in the grid)

mariano-aguero commented:
> #496 is great
> 
> #413 I think that by "Token address" I meant "Wrapped collateral address" or "WMT Address" (or I may have meant to use "Collateral Address" for that... we changed some wording here and there, so I could be wrong, it was called WMT too). Anyways: I don't know if it's possible to implement. We can do it later, maybe?

Fixed

> #411 I think the "Custom Token" option is missing from the "Collateral" filter.
> 

Fixed

gnosis-info commented:
Travis automatic deployment:
 https://pr497--conditionaltokensexplorer.review.gnosisdev.com

gnosis-info commented:
Travis automatic deployment:
 https://pr497--conditionaltokensexplorer.review.gnosisdev.com

mariano-aguero commented:
> @mariano-aguero ,
> I also have verified the changes. Great work! However, here are some notes:
> 
> 1. #411 - Hiding the filters' column does not clear Wrapped collateral filtering

Fixed

> 2. creation date filter shows positions that were created on 'the selected day - 1'
>    ![creation date](https://user-images.githubusercontent.com/70885163/96867303-30f4a280-1475-11eb-84f2-eedbdaf6ad9c.jpg)

Fixed

> 3. validation message is displayed when clear out any filter field (both or any of them - a user cannot start a search when the validation message is displayed)

Fixed

>    ![clear out](https://user-images.githubusercontent.com/70885163/96867312-3520c000-1475-11eb-9f56-fbe16a8d39a0.jpg)
> 4. It seems to me that search by a custom collateral symbol does not work: see the video https://drive.google.com/file/d/16RCYVtBS93JqkKvvfMlIKWXFi9pCTHDv/view

You can't search by custom tokens, only defaults

> 5. the issue is not exactly related to the current ticket, but it is worth fixing: #453 (incorrect number of positions are displayed in the grid)

Already exist an issue for this, so will be fixed in that issue. Is really complex

gnosis-info commented:
Travis automatic deployment:
 https://pr497--conditionaltokensexplorer.review.gnosisdev.com

hpmaxi commented on src/components/filters/DateFilter/index.tsx:
```diff
@@ -90,13 +90,13 @@ export const DateFilter: React.FC<Props> = (props) => {
   const emptyValues = React.useMemo(() => !from && !to, [from, to])
   const validDates = React.useMemo(() => {
-    if (fromDate && fromDate.current && toDate && toDate.current) {
+    if (from && to && fromDate && fromDate.current && toDate && toDate.current) {
       return toDate.current.checkValidity() && fromDate.current.checkValidity()
     } else {
       return true
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [from, to])
+  }, [from, to, fromDate, toDate])
 
```
Everything seems to work smoothly, great job!.
I'm not sure if this is causing the following problem with validation:

![Peek 22-10-2020 15-57](https://user-images.githubusercontent.com/358059/96918125-36a3b580-1480-11eb-8a00-34f5e4c1263c.gif)

Remember that `from` and  `to` are the controlled values, they are in the dependency list to force checking the refs that don't cause re-renderings but allow to call the method of `HTMLInputElement`.

mariano-aguero replied:
I added the FROM and TO because  if the inputs are empty, was still checking and display the warning as Elena says in a previous comment

![Screenshot_20201022_161017](https://user-images.githubusercontent.com/1144028/96918860-54bde580-1481-11eb-8585-caff4236c736.png)

Also, I think is not related to the purpose of this PR, a PR already merged introduced this problem, I just try to give it a shot to solve it

hpmaxi replied:
@mariano-aguero This isn't happening in https://conditional-tokens-explorer.dev.gnosisdev.com/#/positions

mariano-aguero replied:
> @mariano-aguero This isn't happening in https://conditional-tokens-explorer.dev.gnosisdev.com/#/positions

This is a problem related to the DateFilter component, so I will remove the few lines that I changed, and will comment to Elena to recheck this, and in case the problem still exist, create and issue

gabitoesmiapodo reviewed (approved):
🤪

gnosis-info commented:
Travis automatic deployment:
 https://pr497--conditionaltokensexplorer.review.gnosisdev.com

elena-zh commented:
Verified in Dev. 
Several issues are found: 
#501 - Move 'Custom Token' option to the bottom in the Collateral filter
#502 - Positions/Conditions lists: creation date filter shows positions that were created on 'the selected day - 1'
#505 - Positions/Conditions list: Pagination is not reset to the 1st page after filters applying
#506 - Positions list: Search query hags eternally when switch search options

Low priority enhancement is created  - #504
